nucypher/blockchain/eth/deployers.py

```diff
@@ -505,21 +505,60 @@ class StakingEscrowDeployer(BaseContractDeployer, UpgradeableContractMixin, Owna
 
     agency = StakingEscrowAgent
     contract_name = agency.contract_name
+    contract_name_stub = "StakingEscrowStub"
 
     can_be_idle = True
-    preparation_steps = ('contract_deployment', 'dispatcher_deployment')
+    init_steps = ('stub_deployment', 'dispatcher_deployment')
+    preparation_steps = ('contract_deployment', 'dispatcher_retarget')
     activation_steps = ('approve_reward_transfer', 'initialize')
     deployment_steps = preparation_steps + activation_steps
     _proxy_deployer = DispatcherDeployer
 
-    def __init__(self, test_mode: bool = False, *args, **kwargs):
+    def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.__dispatcher_contract = None
 
         token_contract_name = NucypherTokenDeployer.contract_name
         self.token_contract = self.blockchain.get_contract_by_name(registry=self.registry,
                                                                    contract_name=token_contract_name)
-        self.test_mode = test_mode
+        self.policy_manager = self._get_contract(deployer_class=PolicyManagerDeployer)
+        self.adjudicator = self._get_contract(deployer_class=AdjudicatorDeployer)
+        self.worklock = self._get_contract(deployer_class=WorklockDeployer)
+
+    def _get_contract(self, deployer_class) -> VersionedContract:
+        contract_name = deployer_class.contract_name
+        try:
+            proxy_name = deployer_class._proxy_deployer.contract_name
+        except AttributeError:
+            proxy_name = None
+        try:
+            return self.blockchain.get_contract_by_name(registry=self.registry,
+                                                        contract_name=contract_name,
+                                                        proxy_name=proxy_name)
+        except self.registry.UnknownContract:
+            return None
+
+    def _deploy_stub(self, gas_limit: int = None, confirmations: int = 0, **overrides):
+        constructor_kwargs = {
+            "_hoursPerPeriod": self.economics.hours_per_period,
+            "_minLockedPeriods": self.economics.minimum_locked_periods,
+            "_minAllowableLockedTokens": self.economics.minimum_allowed_locked,
+            "_maxAllowableLockedTokens": self.economics.maximum_allowed_locked
+        }
+        constructor_kwargs.update(overrides)
+        constructor_kwargs = {k: v for k, v in constructor_kwargs.items() if v is not None}
+        # Force use of the token address from the registry
+        constructor_kwargs.update({"_token": self.token_contract.address})
```

Comment:
Can this be pushed down a layer or accepted as a parameter?  Why does it need to be updated in `kwargs`?

Reply:
I guess layer below is contract itself. So my opinion this is good place to rewrite predefined constructor values if needed